tests/workflow_task_reported_problems_test.go

```diff
@@ -0,0 +1,360 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	sdkclient "go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/searchattribute"
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/tests/testcore"
+)
+
+type WFTFailureReportedProblemsTestSuite struct {
+	testcore.FunctionalTestBase
+	tv                     *testvars.TestVars
+	initialRetryInterval   time.Duration
+	scheduleToCloseTimeout time.Duration
+	startToCloseTimeout    time.Duration
+
+	shouldFail atomic.Bool
+
+	activityRetryPolicy *temporal.RetryPolicy
+}
+
+func TestWFTFailureReportedProblemsTestSuite(t *testing.T) {
+	s := new(WFTFailureReportedProblemsTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) SetupTest() {
+	s.FunctionalTestBase.SetupTest()
+
+	s.tv = testvars.New(s.T()).WithTaskQueue(s.TaskQueue()).WithNamespaceName(s.Namespace())
+
+	s.initialRetryInterval = 1 * time.Second
+	s.scheduleToCloseTimeout = 30 * time.Minute
+	s.startToCloseTimeout = 15 * time.Minute
+
+	s.activityRetryPolicy = &temporal.RetryPolicy{
+		InitialInterval:    s.initialRetryInterval,
+		BackoffCoefficient: 1,
+	}
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) simpleWorkflow(ctx workflow.Context) (string, error) {
+	if s.shouldFail.Load() {
+		panic("forced-panic-to-fail-wft")
+	}
+	return "done!", nil
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) simpleActivity() (string, error) {
+	return "done!", nil
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) workflowWithActivity(ctx workflow.Context) (string, error) {
+	var ret string
+	err := workflow.ExecuteActivity(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+		ActivityID:             "activity-id",
+		DisableEagerExecution:  true,
+		StartToCloseTimeout:    s.startToCloseTimeout,
+		ScheduleToCloseTimeout: s.scheduleToCloseTimeout,
```

Comment:
Better not set schedule to close timeout, it's a way to add flakiness to tests when CI is slow.